tests/encode_images.rs

```diff
@@ -360,3 +360,19 @@ fn test_multipage_image() {
         assert_eq!(img_decoder.dimensions().unwrap(), (3, 3));
     }
 }
+
+#[test]
+/// verify rows per strip setting
+fn test_rows_per_strip() {
+    let mut file = Cursor::new(Vec::new());
+    let mut img_encoder = TiffEncoder::new(&mut file).unwrap();
+
+    let mut image = img_encoder.new_image::<colortype::Gray8>(100, 100).unwrap();
+    assert_eq!(image.next_strip_sample_count(), 100 * 100);
+    image.rows_per_strip(2).unwrap();
+    assert_eq!(image.next_strip_sample_count(), 2 * 100);
+
+    let img2: Vec<u8> = vec![0; 2 * 100];
+    image.write_strip(&img2[..]).unwrap();
+    assert!(image.rows_per_strip(5).is_err());
```

Comment:
Please add a roundtrip test that tries to decode the written image.

Reply:
I have extended the test with a round-trip